server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -41,6 +42,20 @@ func NewBuildBuddyServer(env environment.Env, sslService *ssl.SSLService) (*Buil
 	}, nil
 }
 
+func (s *BuildBuddyServer) getSanitizedGetInvocationResponse(ctx context.Context, unsanitizedResponse *inpb.GetInvocationResponse) (*inpb.GetInvocationResponse, error) {
```

Comment:
I don't know if this would work but you might be able to just take a pointer to GetInvocationResponse and return an error, re-using the same proto to unmarshal to text, then re-marshal from text.

func (...) redactAPIKey(ctx, rsp *GetInvocationResponse) error {
    txt := proto.MarshalTextString(rsp)
    txt = strings.ReplaceAll(txt, apiKey, "")
    return proto.UnmarshalText(text, rsp)
}

Reply:
Done

Reply:
Also the impl of `UnmarshalText` calls `m.Reset()` before unmarshaling so this should be safe